src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -659,8 +660,9 @@ private static bool IsSignedIntegralOrIntPtrType(ITypeSymbol? type)
             if (castType.Equals(rewrittenConditionalOperation.Type, SymbolEqualityComparer.IncludeNullability))
                 return true;
 
-            if (rewrittenConditionalOperation.Parent is IConversionOperation { IsImplicit: true } implicitConversion &&
-                castType.Equals(implicitConversion.Type, SymbolEqualityComparer.IncludeNullability))
+            if (rewrittenConditionalOperation.Parent is IConversionOperation conditionalParentConversion &&
+                conditionalParentConversion.GetConversion().IsImplicit &&
```

Comment:
How about defining an extension method “IsImplicitConversion” on IConversionOperation? This will likely prevent someone else from making the same mistake when they see IsImplicit and IsImplicitConversion available on it and read through the doc comments on both to choose the appropriate member to invoke.

Reply:
possibly.  though it doesn't help in this case because in a pattern you only see properties :'(